[CC-33024] Remove the deprecated fields of CockroachDB operator (#560)
cockroachdb/helm-charts: CHANGELOG.md

All notable changes to this project will be documented in this file.

## [cockroachdb-parent-25.3.3-preview+1] 2025-10-29

> - Remove the deprecated fields and start using the podTemplate fields.

Reviewer: We should list the deprecated fields, at least at a high level. Like:

```
- Removed the following deprecated fields in favor of `podTemplate`:
    - `crdbCluster.resources.*`
    - `crdbCluster.podLabels.*`
    - `crdbCluster.topologySpreadConstraints.*`
```
etc.

All notable changes to this project will be documented in this file.

## [cockroachdb-parent-25.3.3-preview+1] 2025-10-29

> - Remove the deprecated fields and start using the podTemplate fields. Following fields are deprecated:

Reviewer: ```suggestion
- Remove the following deprecated fields in favor of the corresponding podTemplate fields:
```
